docs/control-structures.rst

```diff
@@ -383,6 +386,58 @@ As a result, the following code is legal, despite being poorly written::
         }
     }
 
+Scoping starting from Version 0.5.0
+-----------------------------------
+
+Starting from version 0.5.0, Solidity will change to the more widespread scoping rules of C99
+(and many other languages): Variables are visible from the point right after their declaration
+until the end of a ``{ }``-block. As an exception to this rule, variables declared in the
+initialization part of a for-loop are only visible until the end of the for-loop.
+
+Variables and other items declared outside of a code block, for example functions, contracts,
+user-defined types, etc., do not change their scoping behaviour. This means you can
+use state variables before they are declared and call functions recursively.
+
+These rules are already introduced now as an experimental feature.
+
+As a consequence, the following examples will compile without warnings (apart from
```

Comment:
Maybe worth mentioning this is not compiling currently due to variable redeclaration.

Reply:
argh, weird, that should not happen.

Reply:
Ah sorry, I used the wrong `solc` version, everything is fine.